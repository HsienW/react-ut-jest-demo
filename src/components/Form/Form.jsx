import React, {Component} from 'react';
// import PropTypes from 'prop-types';

export class Form extends Component {

    constructor() {
        super();
        this.account = React.createRef();
        this.password = React.createRef();
        this.state = {
            message: '',
            account: '',
            password: ''
        };
    }

    handleSubmit = (event) => {
        event.preventDefault();
        this.setState({
            message: 'Sending...',
            account: this.account.current.value,
            password: this.password.current.value
        });
    };

    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <div>Account: <input id="account " type="text" ref={this.account}/></div>
                <p/>
                <div>Password: <input id="password" type="text" ref={this.password}/></div>
                <p/>
                <input type="submit"/>
                <p/>
                <h2 id="message">{this.state.message}</h2>
                <h2 id="show-account">{this.state.account}</h2>
                <h2 id="show-password">{this.state.password}</h2>
            </form>
        );
    }
}


// Form.propTypes = {
// };
