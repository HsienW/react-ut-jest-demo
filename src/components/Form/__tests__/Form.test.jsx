import React from 'react';
import Enzyme from 'enzyme';
import {Form} from '../Form';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

describe('Form component', () => {
    it('should created', () => {
        const component = Enzyme.shallow(<Form/>);
        expect(component).toMatchSnapshot();
    });

    it('should render component', () => {
        const component = Enzyme.mount(<Form/>);
        expect(component.find('form').length).toEqual(1);
        expect(component.find('p').length).toEqual(3);
        expect(component.find('input#account').length).toEqual(1);
        expect(component.find('input#password').length).toEqual(1);
    });
});

describe('handleSubmit() function testing', () => {
    const component = Enzyme.mount(<Form/>);
    const form = component.find(Form);

    it('from data should equal right', () => {
        form.find('input#account').simulate('change', '123');
        form.find('input#password').simulate('change', '456');
        form.find('input#password').simulate('click');
        expect(form.find('h2#message').text()).toEqual('Sending...');
        expect(form.find('h2#account').text()).toEqual('123');
        expect(form.find('h2#password').text()).toEqual('456');
    });
});
