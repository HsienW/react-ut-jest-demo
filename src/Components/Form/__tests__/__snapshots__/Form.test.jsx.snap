// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should created 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          Account: 
          <input
            id="account "
            type="text"
          />
        </div>,
        <p />,
        <div>
          Password: 
          <input
            id="password"
            type="text"
          />
        </div>,
        <p />,
        <input
          type="submit"
        />,
        <p />,
        <h2
          id="message"
        >
          
        </h2>,
        <h2
          id="show-account"
        >
          
        </h2>,
        <h2
          id="show-password"
        >
          
        </h2>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Account: ",
            <input
              id="account "
              type="text"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Account: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "account ",
              "type": "text",
            },
            "ref": Object {
              "current": null,
            },
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Password: ",
            <input
              id="password"
              type="text"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Password: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "password",
              "type": "text",
            },
            "ref": Object {
              "current": null,
            },
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "submit",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "id": "message",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "id": "show-account",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "id": "show-password",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Account: 
            <input
              id="account "
              type="text"
            />
          </div>,
          <p />,
          <div>
            Password: 
            <input
              id="password"
              type="text"
            />
          </div>,
          <p />,
          <input
            type="submit"
          />,
          <p />,
          <h2
            id="message"
          >
            
          </h2>,
          <h2
            id="show-account"
          >
            
          </h2>,
          <h2
            id="show-password"
          >
            
          </h2>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Account: ",
              <input
                id="account "
                type="text"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Account: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "account ",
                "type": "text",
              },
              "ref": Object {
                "current": null,
              },
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Password: ",
              <input
                id="password"
                type="text"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Password: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "password",
                "type": "text",
              },
              "ref": Object {
                "current": null,
              },
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "id": "message",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "id": "show-account",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "id": "show-password",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
